src/Microsoft.Win32.Registry/tests/Registry/Registry_Getvalue_str_str_obj.cs

```diff
@@ -12,17 +12,28 @@ namespace Microsoft.Win32.RegistryTests
     public class Registry_GetValue_str_str_obj : RegistryTestsBase
     {
         [Fact]
-        public static void NegativeTests()
+        public static void NullKeyName_ThrowsArgumentNullException()
         {
-            // Should throw if passed keyName is null
-            Assert.Throws<ArgumentNullException>(
-                () => Registry.GetValue(keyName: null,valueName: null, defaultValue: null));
+            Assert.Throws<ArgumentNullException>("keyName", () => Registry.GetValue(null, null, null));
+        }
 
-            // Passing a string which does NOT start with one of the valid base key names, that should throw ArgumentException.
-            Assert.Throws<ArgumentException>(() => Registry.GetValue("HHHH_MMMM", null, null));
+        public static readonly object[][] ArgumentExceptionTestData =
+        {
+            new object[] { string.Empty }, // Not a valid base key name.
+            new object[] { "HHHH_MMMM" }, // Not a valid base key name.
+            new object[] { "HKEY_CURRENT_USER_FOOBAR" }, // Starts with one of the valid base key names but isn't valid.
+            new object[] { new string('a', 10) }, // String of length 10 that isn't HKEY_USERS.
+            new object[] { new string('a', 17) }, // String of length 17 that isn't HKEY_CURRENT_USER or HKEY_CLASSES_ROOT.
+            new object[] { new string('a', 18) }, // String of length 18 that isn't HKEY_LOCAL_MACHINE.
+            new object[] { new string('a', 19) }, // String of length 19 that isn't HKEY_CURRENT_CONFIG.
+            new object[] { new string('a', 21) } // String of length 21 that isn't HKEY_PERFORMANCE_DATA.
+        };
 
-            // Should throw if passed string which only starts with one of the valid base key names but actually it isn't valid.
-            Assert.Throws<ArgumentException>(() => Registry.GetValue("HKEY_CURRENT_USER_FOOBAR", null, null));
+        [Theory]
+        [MemberData(nameof(ArgumentExceptionTestData))]
+        public static void InvalidKeyName_ThrowsArgumentException(string keyName)
+        {
+            Assert.Throws<ArgumentException>(() => Registry.GetValue(keyName, null, null));
```

Comment:
Wasn't sure if I should check for the paramName here as desktop doesn't specify it, but CoreFX does.